stdlib/cmake/modules/AddSwiftStdlib.cmake

```diff
@@ -2703,8 +2860,104 @@ function(add_swift_target_executable name)
 
         add_custom_command(TARGET ${VARIANT_NAME}
           POST_BUILD
-         COMMAND "codesign" "-f" "-s" "-" "${SWIFT_RUNTIME_OUTPUT_INTDIR}/${VARIANT_NAME}")
-       endif()
+          COMMAND "codesign" "-f" "-s" "-" "${SWIFT_RUNTIME_OUTPUT_INTDIR}/${VARIANT_NAME}")
+      endif()
+
+      list(APPEND THIN_INPUT_TARGETS ${VARIANT_NAME})
     endforeach()
+
+    set(library_subdir "${SWIFT_SDK_${sdk}_LIB_SUBDIR}")
+    if(maccatalyst_build_flavor STREQUAL "ios-like")
+      set(library_subdir "${SWIFT_SDK_MACCATALYST_LIB_SUBDIR}")
+    endif()
+
+    if("${sdk}" STREQUAL "WINDOWS")
+      set(UNIVERSAL_NAME "${SWIFTLIBEXEC_DIR}/${library_subdir}/${name}.exe")
+    else()
+      set(UNIVERSAL_NAME "${SWIFTLIBEXEC_DIR}/${library_subdir}/${name}")
+    endif()
+
+    set(lipo_target "${name}-${sdk}")
+    if("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
+      set(codesign_arg CODESIGN)
+    endif()
+    precondition(THIN_INPUT_TARGETS)
+    _add_swift_lipo_target(SDK
+                             ${sdk}
+                           TARGET
+                             ${lipo_target}
+                           OUTPUT
+                             ${UNIVERSAL_NAME}
+                           ${codesign_arg}
+                           ${THIN_INPUT_TARGETS})
+
+    # Determine the subdirectory where this executable will be installed
+    set(resource_dir_sdk_subdir "${SWIFT_SDK_${sdk}_LIB_SUBDIR}")
+    if(maccatalyst_build_flavor STREQUAL "ios-like")
+      set(resource_dir_sdk_subdir "${SWIFT_SDK_MACCATALYST_LIB_SUBDIR}")
+    endif()
+
+    precondition(resource_dir_sdk_subdir)
+
+    if(sdk STREQUAL WINDOWS AND CMAKE_SYSTEM_NAME STREQUAL Windows)
```

Comment:
A bit unclear why `if("${sdk}" STREQUAL "WINDOWS")` is used above, but this quite different comparison is used on this line, maybe that warrants an explanatory comment?

Reply:
IIRC the quotes and string expansion are unnecessary. CMake is weird.

Reply:
No good reason. The two mean the same thing AFAIK.

Reply:
`if("${sdk}" STREQUAL "WINDOWS")` is the most foolproof version that always works, at the cost of a little punctuation noise, whereas this unquoted form sometimes fails on some platforms, who knows why. 🤷